backend/elasticsearch/src/test/java/org/hibernate/search/backend/elasticsearch/search/projection/impl/DistanceToFieldSearchProjectionTest.java

```diff
@@ -30,10 +32,14 @@ public void projection_script() {
 
 		JsonObject requestBody = new JsonObject();
 
-		SearchProjectionExecutionContext searchProjectionExecutionContext =
-				elementCollector.toSearchProjectionExecutionContext();
+		SessionContextImplementor sessionContext = createMock( SessionContextImplementor.class );
 
+		resetAll();
+		replayAll();
+		SearchProjectionExecutionContext searchProjectionExecutionContext =
+				elementCollector.toSearchProjectionExecutionContext( sessionContext );
 		projection.contributeRequest( requestBody, searchProjectionExecutionContext );
+		verifyAll();
```

Comment:
Why do we need all this noise? Doesn't look like an improvement for the readability of the test?

Wouldn't a stub `SessionContext` be better?

Reply:
This "noise" is necessary for EasyMock mocks to be in the correct state when they are used. Without it, the mock would be in "record" mode during the test, which is not the way mocks are supposed to be used. As an additional benefit, putting the mock in the correct state allows us to check that no method is called on the session context, in other words that `toSearchProjectionExecutionContext` does not rely on the session context. Not really necessary, but as I said it's not the main reason.

As to whether coding empty interfaces yourself is better than using a mocking library, well... I personally don't want to have to change this test whenever we add a method to `SessionContextImplementor`, but YMMV.